R/zchunk_LA161.Cstorage_china.R

```diff
@@ -0,0 +1,113 @@
+      # Calculate cumulative sum, then filter to the quantiles
+      mutate(Cumul_MtC = cumsum(MtC)) %>%
+      # use "type=1" to indicate that we want to return only exact matches to the data in the table.
+      # only take the (0, 0.2, 0.4, 0.6, 0.8, 1) points in the quartiles
+      filter(Cumul_MtC %in% quantile(Cumul_MtC, c(0, 0.2, 0.4, 0.6, 0.8, 1), type = 1)) %>%
+      # this will create grade equal to grade 1 for 0, grade 2 for 0.2, etc.
+      # from the cumulative totals, return to the marginal quantities associated with each grade
+      mutate(grade = paste("grade", 1:6, sep = " "),
+             MtC = Cumul_MtC - lag(Cumul_MtC, 1, 0)) %>%
+      ungroup %>%
+      # Return from the list to a data frame with all necessary data
+      select(province.name, grade, Cost_1990USDtC, MtC) %>%
+      # Add cost grades to provinces without data
+      bind_rows(CCS_China_add) %>%
+      # Setting a minimum cost of 0 on CO2 storage and transport projects
+      mutate(Cost_1990USDtC = pmax(Cost_1990USDtC, 0)) %>%
+      # Manually changing cost curves to avoid grades 2 or 3 being 0. Value determined by linear interpolation. If the data changes in the future, will need to change this
+      mutate(Cost_1990USDtC = replace(Cost_1990USDtC, province.name == "Jilin" & grade == "grade 2", 4.625),
+             Cost_1990USDtC = replace(Cost_1990USDtC, province.name == "Xinjiang" & grade == "grade 2", 4.3),
+             Cost_1990USDtC = replace(Cost_1990USDtC, province.name == "Yunnan" & grade == "grade 2", 5.74),
+             Cost_1990USDtC = replace(Cost_1990USDtC, province.name == "Yunnan" & grade == "grade 3", 11.48)) ->
+      L161.Cstorage_province
+
+    # Extending the plateau (bin 2) of the cost curve using additional resource data
+    L161.Cstorage_province %>%
+      filter(grade == "grade 2") %>%
+      left_join_error_no_match(Unallocated_CStorage, by = "province.name") %>%
+      # Convert to MtC for additional resource
+      mutate(add_storage = MtCO2 * (1 / emissions.CONV_C_CO2),
+             # Gansu is a special case, we are not adding storage
+             add_storage = replace(add_storage, province.name == "Gansu", 0),
+             MtC = MtC + add_storage) %>%
+      select(province.name, grade, Cost_1990USDtC, MtC) ->
+      L161.Cstorage_province_g2
+
+    L161.Cstorage_province %>%
+      filter(grade != "grade 2") %>%
+      bind_rows(L161.Cstorage_province_g2) %>%
+      arrange(province.name, grade) %>%
+      map_province_name(province_names_mappings, "province", FALSE) %>%
+      select(province, province.name, grade, Cost_1990USDtC, MtC) ->
+      L161.Cstorage_province
+
+    # ===================================================
+    # Produce outputs
+    L161.Cstorage_province %>%
+      add_title("CO2 storage curves by province and grade") %>%
+      add_units("MtC; 1990 USD per tC") %>%
+      add_comments("Cumulative MtC calculated by grid region, then filtered to quartiles") %>%
```

Comment:
"Cumulative MtC calculated by province, then filtered to quartiles"

Reply:
get